File: src/main/kotlin/org/daonators/client/request/AuthRequest.kt
package org.daonators.client.request

import br.com.simpli.model.LanguageHolder
import br.com.simpli.tools.Validator
import io.swagger.v3.oas.annotations.media.Schema
import org.daonators.client.auth.AuthProcess
import org.daonators.exception.response.BadRequestException

/**
 * Authentication Request Model
 * @author Simpli CLI generator
 */
class AuthRequest(var email: String?, var password: String?) {
    fun validate(lang: LanguageHolder) {
        if (email.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("Email"))
        }

        if (!Validator.isEmail(email)) {
            throw BadRequestException(lang.isNotAValidEmail("Email"))
        }

        if (password.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("Password"))
        }
    }

    @Schema(hidden = true)
    fun toToken() = AuthProcess.requestToToken(this)
}

File: src/main/webapp/src/model/request/AuthRequest.ts
/**
 * AuthRequest
 * @author Simpli CLI generator
 */
import {$} from '@/facade'
import {Request, ResponseExclude} from '@simpli/serialized-request'
import {AuthResponse} from '@/model/response/AuthResponse'

/* TODO: review generated class */
export class AuthRequest {
  email: string | null = null

  @ResponseExclude()
  password: string | null = null

  /**
   * Gets the user authentication
   */
  static async authenticate() {
    return await Request.get(`/client/auth`)
      .name('authenticate')
      .as(AuthResponse)
      .getData()
  }

  /**
   * Submits the user authentication
   */
  async signIn() {
    const request = $.utils.clone(this)
    request.password = $.utils.sha256(this.password)

    return await Request.post(`/client/auth`, request)
      .name('signIn')
      .delay(1000)
      .as(AuthResponse)
      .getData()
  }
}
